crates/sargon/src/system/sargon_os/sargon_os_entities_linked_to_factor_source.rs

```diff
@@ -0,0 +1,317 @@
+use crate::prelude::*;
+
+impl SargonOS {
+    /// Returns the entities linked to a given `FactorSource`, either on the current `Profile` or a specific one.
+    pub async fn entities_linked_to_factor_source(
+        &self,
+        factor_source: FactorSource,
+        profile_to_check: ProfileToCheck,
+    ) -> Result<EntitiesLinkedToFactorSource> {
+        let integrity = self.integrity(factor_source.clone()).await?;
+        match profile_to_check {
```

Comment:
prefer expressions:

```rust
        let profile = match profile_to_check {
            ProfileToCheck::Current => self.profile()?
            ProfileToCheck::Specific(specific_profile) => specific_profile
        };

profile
                .current_network()?
                .entities_linked_to_factor_source(factor_source, integrity),
```

Reply:
this doesn't compile because of multiple errors about borrowed values not living long enough, and in order to fix it I need to add more variables which I think complicates the code more.